py/desispec/preproc.py

```diff
@@ -739,18 +745,86 @@ def preproc(rawimage, header, primary_header, bias=True, dark=True, pixflat=True
         bkg = _background(image,header)
         image -= bkg
 
-
     img = Image(image, ivar=ivar, mask=mask, meta=header, readnoise=readnoise, camera=camera)
 
     #- update img.mask to mask cosmic rays
-
     if not nocosmic :
         cosmics.reject_cosmic_rays(img,nsig=cosmics_nsig,cfudge=cosmics_cfudge,c2fudge=cosmics_c2fudge)
+        mask = img.mask
 
-    if remove_scattered_light :
+
+    xyset = None
+
+    if use_variance_model  :
+
+        psf = None
         if psf_filename is None :
             psf_filename = cfinder.findfile("PSF")
         xyset = read_xytraceset(psf_filename)
+
+        """
+        # use the actual PSF model for 1D projection
+        # this is too slow for now
+        import fitsio
+        import specter.psf
+        head  = fitsio.read_header(psf_filename)
+        if "PSFTYPE" in head :
+            psftype=head["PSFTYPE"].strip()
+            if psftype=="GAUSS-HERMITE" :
+                psf = specter.psf.GaussHermitePSF(psf_filename)
+            elif psftype=="SPOTGRID" :
+                psf = specter.psf.SpotGridPSF(psf_filename)
+        """
```

Comment:
Maybe remove this section that is commented-via-quotes?

Reply:
done